### ISSUE TITLE
SnipMate interacts poorly with Vim autoformatting

### ISSUE THREAD

**Issue by llinfeng** [garbas/vim-snipmate#173]
I admit it is bad habit to use long variable names. This leads to long variable names and auto-completion becomes a must for me.

Snipmate does not like my autocompletion. I am not using anything fancy, but just plain `<C-N>` and `<C-P>`. However, whenever I touch these two key-strokes, snipmate stops to function after that point. It completes the word, but will distort the rest of the line at which it automatically updates. Here is the snippet that I have trouble: 

```
    eststo: qui areg ${1:LHS} \$${2:Dfdn_traits}, absorb(${3:FE}) cluster(${4:clustering_variable}) 
    foreach j of varlist $2 {
        disp "\`j'"
            eststo: qui  areg $1  \`j', absorb($3) cluster($4) 
    }
    esttab using $Tex/05_ScreeningADA_yearlyIV_monthly_FE.tex, replace se
    eststo clear
```

Another problem is that: Snipmate does not like auto-linebreak, when it "syncs" the `$1`'s. When I reach 80 characters at the first line, with my long variable name, I will get a hard wrap at that point. When I reach such point, the line where $4 is located is also distorted in a way that the "filling-in" of characters from $4 defined above just stops, leaving an incomplete variable name in this line.

Both of the two obstacles distorts the experience with using snipmate.

As a way around, I now type the full variable names and have turned off the hard-wrapping at 80 characters.

Thanks again for making Snipmate possible for Vimmers.

**Comment by ajzafar** [comment]
@llinfeng I am no longer able to reproduce your first problem in the just-pushed master.

**Comment by llinfeng** [comment]
Yeah, I confirm that the snipmate is working well now with `<C-N>` and `<C-P>` expansions. Thank you for making this possible.

All the best, 

-Linfeng
